### PR TITLE
Add new operator SubtractionNodeByLabelsScan

### PR DESCRIPTION
**Pull request by pontusmelke**

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-cypher-982.surge.sh

Updated pages:
- [Operators](https://neo4j-docs-cypher-982.surge.sh/cypher-manual/5/planning-and-tuning/operators)
- [Operators in detail](https://neo4j-docs-cypher-982.surge.sh/cypher-manual/5/planning-and-tuning/operators/operators-detail)

File: modules/ROOT/pages/planning-and-tuning/operators/operators-detail.adoc [hunk 1]
```diff
@@ -1266,6 +1266,87 @@ Total database accesses: 13
 
 ======
 
+[query-plan-subtraction-node-by-labels-scan]]
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/planning-and-tuning/operators/operators-detail.adoc]:
```suggestion
[role=label--new-5.21]
[[query-plan-subtraction-node-by-labels-scan]]
```

File: modules/ROOT/pages/planning-and-tuning/operators/operators-detail.adoc [hunk 2]
```diff
@@ -1266,6 +1266,87 @@ Total database accesses: 13
...
+== Subtraction Node By Labels Scan
+// SubtractionNodeByLabelsScan
+// New in 5.21
+
+The `SubtractionNodeByLabelsScan` operator fetches all nodes that have all of the first set of provided labels and none of the second provided set of labels from the node label index.
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/planning-and-tuning/operators/operators-detail.adoc:1275]:
```suggestion
The `SubtractionNodeByLabelsScan` operator fetches all nodes that have all of the first set of provided labels and none of the second provided set of labels from the node label index.
In the example below, `SubtractionNodeByLabelsScan` retrieves all nodes that have the `Location` label but do not have the `Country` label.
```

File: modules/ROOT/pages/planning-and-tuning/operators/operators-detail.adoc [hunk 3]
```diff
@@ -1266,6 +1266,87 @@ Total database accesses: 13
...
+| |                            +----+--------------------------------------+----------------+------+---------+----------------+                        |           |                     |
+| +SubtractionNodeByLabelsScan |  1 | locationNotCountry:Location&!Country |              7 |   10 |      13 |            248 |                    2/0 |     3.081 | Fused in Pipeline 0 |
++------------------------------+----+--------------------------------------+----------------+------+---------+----------------+------------------------+-----------+---------------------+
+
+Total database accesses: 13, total allocated memory: 312
+----
+
+======
+
+[role=label--new-5.17]
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/planning-and-tuning/operators/operators-detail.adoc]:
```suggestion
[role=label--new-5.21]
```

**Review by JPryce-Aklundh — COMMENTED** [review]
Looks good - just a few suggestions.

File: modules/ROOT/pages/planning-and-tuning/operators/index.adoc [hunk 4]
```diff
@@ -767,6 +767,13 @@ It uses a bidirectional breadth-first search (BFS) algorithm, which performs two
 |
 |
 
+| xref::planning-and-tuning/operators/operators-detail.adoc#query-plan-subtraction-node-by-labels-scan[SubtractionNodeByLabelsScan]
+| Fetches all nodes that have all of the first set of provided labels and none of the second provided set of labels from the node label index.
+| label:yes[]
+|
+|
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/planning-and-tuning/operators/index.adoc]:
```suggestion
| label:new[Introduced in 5.21]
```

File: modules/ROOT/pages/planning-and-tuning/operators/index.adoc [hunk 5]
```diff
@@ -767,6 +767,13 @@ It uses a bidirectional breadth-first search (BFS) algorithm, which performs two
 |
 |
 
+| xref::planning-and-tuning/operators/operators-detail.adoc#query-plan-subtraction-node-by-labels-scan[SubtractionNodeByLabelsScan]
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/planning-and-tuning/operators/index.adoc:776]: Should not Partitioned Subtraction Node By Labels Scan also be added?

File: modules/ROOT/pages/planning-and-tuning/operators/operators-detail.adoc [hunk 6]
```diff
@@ -1266,6 +1266,87 @@ Total database accesses: 13
...
+// New in 5.21
+
+The `PartitionedSubtractionNodeByLabelsScan` is a variant of the xref:planning-and-tuning/operators/operators-detail.adoc#query-plan-subtraction-node-by-labels-scan[`SubtractionNodeByLabelsScan`] operator used by the xref:planning-and-tuning/runtimes/concepts.adoc#runtimes-parallel-runtime[parallel runtime].
+It allows the index to be partitioned into different segments where each segment can be scanned independently in parallel.
+======
+
+.Query
+[source,cypher]
+----
+CYPHER runtime=parallel
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/planning-and-tuning/operators/operators-detail.adoc:1324]:
```suggestion
CYPHER runtime=parallel
PROFILE
```

**Review by JPryce-Aklundh — COMMENTED** [review]
Forgot some things :)

File: modules/ROOT/pages/planning-and-tuning/operators/index.adoc (hunk 5 above)
Comment (pontusmelke) [reply modules/ROOT/pages/planning-and-tuning/operators/index.adoc:776]: yes, I'll fix that

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-cypher-982.surge.sh

Updated pages:
- [Operators](https://neo4j-docs-cypher-982.surge.sh/cypher-manual/5/planning-and-tuning/operators)
- [Operators in detail](https://neo4j-docs-cypher-982.surge.sh/cypher-manual/5/planning-and-tuning/operators/operators-detail)

**Comment by neo-technology-commit-status-publisher** [comment]
Thanks for the documentation updates.

The preview documentation has now been torn down - reopening this PR will republish it.

File: modules/ROOT/pages/planning-and-tuning/operators/index.adoc (hunk 5 above)
Comment (pontusmelke) [reply modules/ROOT/pages/planning-and-tuning/operators/index.adoc:776]: Fixed now

**Review by JPryce-Aklundh — APPROVED** [review]

**Pull request merged**